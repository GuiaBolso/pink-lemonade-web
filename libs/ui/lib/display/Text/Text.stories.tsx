import React from 'react';
import { Text } from '.';

export default {
  title: 'Display/Text',
};

export const display00 = () => <Text variant="display-00">display-00</Text>;
export const display01 = () => <Text variant="display-01">display-01</Text>;
export const display02 = () => <Text variant="display-02">display-02</Text>;
export const heading01 = () => <Text variant="heading-01">heading-01</Text>;
export const heading02 = () => <Text variant="heading-02">heading-02</Text>;
export const heading03 = () => <Text variant="heading-03">heading-03</Text>;
export const heading04 = () => <Text variant="heading-04">heading-04</Text>;
export const heading05 = () => <Text variant="heading-05">heading-05</Text>;
export const heading06 = () => <Text variant="heading-06">heading-06</Text>;
export const body01 = () => <Text variant="body-01">body-01</Text>;
export const body02 = () => <Text variant="body-02">body-02</Text>;
export const subtitle01 = () => <Text variant="subtitle-01">subtitle-01</Text>;
export const subtitle02 = () => <Text variant="subtitle-02">subtitle-02</Text>;
export const buttonText = () => <Text variant="button-text">button-text</Text>;
export const caption = () => <Text variant="caption">caption</Text>;
export const overline = () => <Text variant="overline">overline</Text>;
export const link = () => <Text variant="link">link</Text>;
