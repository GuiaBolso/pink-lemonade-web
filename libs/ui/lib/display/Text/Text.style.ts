import styled from '@emotion/styled';
import { css } from '@emotion/core';

type TextProps = {
  variant: string;
  as: React.ElementType;
};

const variants: object = {
  'display-00': ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize80};
        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.light};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'display-01': ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize64};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxxl};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.bold};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'display-02': ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize48};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxl};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.bold};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'heading-01': ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize42};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xl};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.bold};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'heading-02': ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize32};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.lg};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.bold};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'heading-03': ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize24};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.md};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.bold};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'heading-04': ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize20};

        color: ${theme?.colors?.neutral.darkest};
        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
      `,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.sm};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.bold};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'heading-05': ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize18};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.semibold};
        line-height: ${theme?.type?.lineHeight?.rule2};
      `,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xs};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.semibold};
        line-height: ${theme?.type?.lineHeight?.rule1};
      `,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'heading-06': ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize14};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.semibold};
        line-height: ${theme?.type?.lineHeight?.rule2};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'body-01': ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize16};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'body-02': ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize14};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'subtitle-01': ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize18};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule3};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'subtitle-02': ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize16};

        font-family: ${theme?.type?.fontFamily?.brand};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule2};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'button-text': ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize16};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  caption: ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize12};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  overline: ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize12};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
        text-transform: uppercase;
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  link: ({ theme }) => {
    const themeName = {
      connect: () => css`
        --fontSize: ${theme?.type?.sizes?.fontSize16};

        color: ${theme?.colors?.neutral?.darkest};
        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.semibold};
        line-height: ${`-webkit-${theme?.type?.lineHeight?.rule2}`};
        line-height: ${`-moz-${theme?.type?.lineHeight?.rule2}`};
        line-height: ${theme?.type?.lineHeight?.rule2};
        text-decoration: underline;

        &:hover {
          color: ${theme?.colors?.brand?.secondary?.dark};
        }

        &:visited {
          color: ${theme?.colors?.brand?.primary?.default};
        }
      `,
      guiaBolso: () => null,
    };

    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'body-medium': ({ theme }) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xs};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule2};
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'body-small': ({ theme }) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxs};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule2};
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'subtitle-medium': ({ theme }) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxs};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'subtitle-small': ({ theme }) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxxs};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule2};
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'overline-medium': ({ theme }) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxs};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
        text-transform: uppercase;
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'overline-small': ({ theme }) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxxs};

        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.regular};
        line-height: ${theme?.type?.lineHeight?.rule1};
        text-transform: uppercase;
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'link-medium': ({ theme }) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xs};

        color: ${theme?.colors?.brand?.secondary?.default};
        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.semibold};
        line-height: ${theme?.type?.lineHeight?.rule2};
        text-decoration: underline;
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },

  'link-small': ({ theme }) => {
    const themeName = {
      connect: () => null,
      guiaBolso: () => css`
        --fontSize: ${theme?.type?.sizes?.xxs};

        color: ${theme?.colors?.brand?.secondary?.default};
        font-family: ${theme?.type?.fontFamily?.neutral};
        font-size: var(--fontSize);
        font-weight: ${theme?.type?.fontWeight?.semibold};
        line-height: ${theme?.type?.lineHeight?.rule2};
        text-decoration: underline;
      `,
    };
    return (themeName[theme?.name] || themeName.guiaBolso)();
  },
};

export const Text = styled.span<TextProps>`
  margin: 0;
  padding: 0;

  /* stylelint-disable-next-line */
  ${props => variants[props.variant]}
`;
