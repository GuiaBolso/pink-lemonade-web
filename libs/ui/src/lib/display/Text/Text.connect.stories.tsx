import React from 'react';
import { Text } from '.';
import { Theme } from '../../general';
import { connect } from '../../../designSystem/tokens';

export default {
  title: 'Display/Text/Connect',
};

export const display00 = () => (
  <Theme theme={connect}>
    <Text variant="display-00">display-00</Text>
  </Theme>
);
export const display01 = () => (
  <Theme theme={connect}>
    <Text variant="display-01">display-01</Text>
  </Theme>
);
export const display02 = () => (
  <Theme theme={connect}>
    <Text variant="display-02">display-02</Text>
  </Theme>
);
export const heading01 = () => (
  <Theme theme={connect}>
    <Text variant="heading-01">heading-01</Text>
  </Theme>
);
export const heading02 = () => (
  <Theme theme={connect}>
    <Text variant="heading-02">heading-02</Text>
  </Theme>
);
export const heading03 = () => (
  <Theme theme={connect}>
    <Text variant="heading-03">heading-03</Text>
  </Theme>
);
export const heading04 = () => (
  <Theme theme={connect}>
    <Text variant="heading-04">heading-04</Text>
  </Theme>
);
export const heading05 = () => (
  <Theme theme={connect}>
    <Text variant="heading-05">heading-05</Text>
  </Theme>
);
export const heading06 = () => (
  <Theme theme={connect}>
    <Text variant="heading-06">heading-06</Text>
  </Theme>
);
export const body01 = () => (
  <Theme theme={connect}>
    <Text variant="body-01">body-01</Text>
  </Theme>
);
export const body02 = () => (
  <Theme theme={connect}>
    <Text variant="body-02">body-02</Text>
  </Theme>
);
export const subtitle01 = () => (
  <Theme theme={connect}>
    <Text variant="subtitle-01">subtitle-01</Text>
  </Theme>
);
export const subtitle02 = () => (
  <Theme theme={connect}>
    <Text variant="subtitle-02">subtitle-02</Text>
  </Theme>
);
export const buttonText = () => (
  <Theme theme={connect}>
    <Text variant="button-text">button-text</Text>
  </Theme>
);
export const caption = () => (
  <Theme theme={connect}>
    <Text variant="caption">caption</Text>
  </Theme>
);
export const overline = () => (
  <Theme theme={connect}>
    <Text variant="overline">overline</Text>
  </Theme>
);
export const link = () => (
  <Theme theme={connect}>
    <Text variant="link" component="a" href="#">
      link
    </Text>
  </Theme>
);
