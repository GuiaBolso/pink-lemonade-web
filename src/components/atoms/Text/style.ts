import styled, { css } from 'styled-components';
import tokens from '../../../tokens';

const variants: any = {
  caption: () => css`
    font-size: ${tokens.fontSize('xxs')};
    font-weight: ${tokens.get('type.fontWeight.bold')};
    line-height: ${tokens.get('type.lineHeight.xs')};
    text-transform: uppercase;
  `,

  subtitle: () => css`
    font-size: ${tokens.fontSize('xs')};
    font-weight: ${tokens.get('type.fontWeight.normal')};
    line-height: ${tokens.get('type.lineHeight.s')};
  `,

  'body-1': () => css`
    font-size: ${tokens.fontSize('m')};
    font-weight: ${tokens.get('type.fontWeight.normal')};
    line-height: ${tokens.get('type.lineHeight.l')};
  `,

  'body-2': () => css`
    font-size: ${tokens.fontSize('s')};
    font-weight: ${tokens.get('type.fontWeight.normal')};
    line-height: ${tokens.get('type.lineHeight.m')};
  `,

  'title-1': () => css`
    font-size: ${tokens.fontSize('l')};
    font-weight: ${tokens.get('type.fontWeight.bold')};
    line-height: ${tokens.get('type.lineHeight.l')};
  `,

  'title-2': () => css`
    font-size: ${tokens.fontSize('m')};
    font-weight: ${tokens.get('type.fontWeight.semiBold')};
    line-height: ${tokens.get('type.lineHeight.l')};
  `,

  'primary-button': () => css`
    font-size: ${tokens.fontSize('m')};
    font-weight: ${tokens.get('type.fontWeight.normal')};
    line-height: ${tokens.get('type.lineHeight.l')};
  `,

  'seconday-button': () => css`
    font-size: ${tokens.fontSize('m')};
    font-weight: ${tokens.get('type.fontWeight.normal')};
    line-height: ${tokens.get('type.lineHeight.l')};
  `,

  'tertiary-button': () => css`
    font-size: ${tokens.fontSize('m')};
    font-weight: ${tokens.get('type.fontWeight.semiBold')};
    line-height: ${tokens.get('type.lineHeight.l')};
  `,

  'helper-text': () => css`
    font-size: ${tokens.fontSize('xs')};
    font-weight: ${tokens.get('type.fontWeight.normal')};
    line-height: ${tokens.get('type.lineHeight.s')};
  `,

  'input-label': () => css`
    font-size: ${tokens.fontSize('m')};
    font-weight: ${tokens.get('type.fontWeight.semiBold')};
    line-height: ${tokens.get('type.lineHeight.l')};
  `,

  'input-text': () => css`
    font-size: ${tokens.fontSize('m')};
    font-weight: ${tokens.get('type.fontWeight.normal')};
    line-height: ${tokens.get('type.lineHeight.l')};
  `,

  'big-number': () => css`
    font-size: ${tokens.fontSize('xl')};
    font-weight: ${tokens.get('type.fontWeight.bold')};
    line-height: ${tokens.get('type.lineHeight.xl')};
  `,
};

export const Text = styled.span<{ variant: string }>`
  font-family: ${tokens.get('type.fontFamily.system')};
  ${props => variants[props.variant]};
`;
