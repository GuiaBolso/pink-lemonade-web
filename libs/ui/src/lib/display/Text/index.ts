import { Text, VariantProps } from './Text';

export { Text, VariantProps };
