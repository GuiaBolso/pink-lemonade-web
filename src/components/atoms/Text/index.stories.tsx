import React from 'react';
import Text from '.';

export default {
  title: 'Atoms/Text',
  component: Text,
};

export const caption = () => (
  <Text component="p" variant="caption">
    Hello Text
  </Text>
);
export const subtitle = () => (
  <Text component="p" variant="subtitle">
    Hello Text
  </Text>
);
export const body1 = () => (
  <Text component="p" variant="body-1">
    Hello Text
  </Text>
);
export const body2 = () => (
  <Text component="p" variant="body2">
    Hello Text
  </Text>
);
export const title1 = () => (
  <Text component="p" variant="title1">
    Hello Text
  </Text>
);
export const title2 = () => (
  <Text component="p" variant="title2">
    Hello Text
  </Text>
);
export const primaryButton = () => (
  <Text component="p" variant="primary-button">
    Hello Text
  </Text>
);
export const secondaryButton = () => (
  <Text component="p" variant="secondary-button">
    Hello Text
  </Text>
);
export const tertiaryButton = () => (
  <Text component="p" variant="tertiary-button">
    Hello Text
  </Text>
);
export const helperText = () => (
  <Text component="p" variant="helper-text">
    Hello Text
  </Text>
);
export const inputLabel = () => (
  <Text component="p" variant="input-label">
    Hello Text
  </Text>
);
export const inputText = () => (
  <Text component="p" variant="input-text">
    Hello Text
  </Text>
);
export const bigNumber = () => (
  <Text component="p" variant="big-number">
    Hello Text
  </Text>
);
