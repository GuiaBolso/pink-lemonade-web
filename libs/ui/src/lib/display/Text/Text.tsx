import React, { memo } from 'react';
import * as Styled from './Text.style';

export type VariantProps =
  | 'display-00'
  | 'display-01'
  | 'display-02'
  | 'heading-01'
  | 'heading-02'
  | 'heading-03'
  | 'heading-04'
  | 'heading-05'
  | 'heading-06'
  | 'body-01'
  | 'body-02'
  | 'subtitle-01'
  | 'subtitle-02'
  | 'button-text'
  | 'caption'
  | 'overline'
  | 'link'
  | 'body-medium'
  | 'body-small'
  | 'subtitle-medium'
  | 'subtitle-small'
  | 'overline-medium'
  | 'overline-small'
  | 'link-medium'
  | 'link-small';

export type TextProps = React.ComponentProps<'span'> & {
  /**
   * Component tag that will be rendered
   */
  component?: React.ElementType; // Component tag that will be rendered (div, p, strong, etc)
  /**
   * Text or Component that will be rendered (string, number, other component)
   */
  children: React.ReactNode;
  /**
   * Variation of style that the component will assume (caption, subtitle-01, link, etc)
   */
  variant?: VariantProps;
};

export const Text = ({
  children,
  component = 'span',
  variant = 'body-01',
  ...rest
}: TextProps) => {
  return (
    <Styled.Text as={component} variant={variant} data-testid="text" {...rest}>
      {children}
    </Styled.Text>
  );
};

export default memo(Text);
