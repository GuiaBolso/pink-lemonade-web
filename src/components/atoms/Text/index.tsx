import React from 'react';
import * as Styled from './style';

interface TextProps {
  /**
   * Component tag that will be rendered
   */
  component?: React.ElementType; // Component tag that will be rendered (div, p, strong, etc)
  /**
   * Text or Component that will be rendered (string, number, other component)
   */
  children: React.ReactNode;
  /**
   * Variation of style that the component will assume (caption, subtitle, title-1, etc)
   */
  variant?: string;
}

const Text = ({
  children,
  component = 'span',
  variant = 'body-1',
}: TextProps) => {
  return (
    <Styled.Text as={component} variant={variant}>
      {children}
    </Styled.Text>
  );
};

export default Text;
