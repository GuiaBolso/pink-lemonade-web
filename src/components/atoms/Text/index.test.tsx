import React from 'react';
import { shallow } from 'enzyme';
import Text from '.';

describe('Text', () => {
  it('Renders Text Hello', () => {
    const element = shallow(
      <Text component="p" variant="body-1">
        Hello
      </Text>
    );
    expect(element).toMatchSnapshot();
  });
});
