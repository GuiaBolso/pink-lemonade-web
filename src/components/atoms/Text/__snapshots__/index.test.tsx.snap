// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Renders Text Hello 1`] = `ShallowWrapper {}`;
